### ISSUE TITLE
[Issue Report]: Cannot get gold from stash

### ISSUE THREAD

**Issue by locksoft** [diasurgical/DevilutionX#6988]
### Operating System

Nintendo Switch

### DevilutionX version

1.5.2

### Describe

Once I’m in the stash, the “-“ button (game calls it “select”) opens the form to type in the number of gold coins to move to inventory. The system keyboard correctly opens but the digits are not typed in the game form, so it’s not possible to get them.

### To Reproduce

1 - Talk to Gillian and choose “access stash” (I don’t know the correct english menu name, as I’m playing the game in another language);
2 - press the “-“ (“select”) button;
3 - System keyboard appears, but nothing typed fills the text field.

### Expected Behavior

The typed digits should appears in the game text field.

**Comment by Trihedraf** [comment]
While it is a bug, you don't ever need to pull gold out of the stash unless you are giving it to someone. The vendors will pull out of the stash directly.

**Comment by locksoft** [comment]
Yes, you’re right. I just reported it as it wasn’t working, thanks all. :)

**Comment by locksoft** [comment]
> @locksoft FYI, there is an issue with the 3DS on-screen keyboard that makes using it for splitting/withdrawing gold infeasible. The application actually pauses the main thread while the keyboard is open. The same thing appears to happen in Ryujinx. Can you tell me if the game freezes up when the keyboard is open on the Switch console?

On Switch the game keeps running. I can’t tell for sure at Gillian because nothing happens in town but, if I try to split the gold in the inventory while I’m in the dungeon, monsters keep on attacking.
Also this form is a bit different from the Gillian’s one: here I can insert the numbers in the form, but nothing happens. It’s like if it’s missing an OK or CANCEL button. The form stays open while I can keep playing normally. The only way to dismiss it is to open the menu.

**Comment by locksoft** [comment]
I've made a quick video showing all the issues I've encountered so far in the Switch version of DevilutionX, i think it's easier than writing a wall of text. :)
Please, keep in mind I'm italian, so my english is not really good; and I had the phone in front of the Switch, so I didn't see very well what I was doing.

But I hope it's clear enough:

http://gofile.me/6RvCQ/RKbmcwsYK

**Comment by AJenbo** [comment]
Arh I see you are also reporting a secound issue here with the lower right cornor in the inventory not working correctly with picking up an item that is 1x1.

**Comment by locksoft** [comment]
> I edited your message to have the video directly on GitHub instead of some file host site.

Thank you. It was my NAS. :)

> I see you are also reporting a second issue here with the lower right corner in the inventory not working correctly with picking up an item that is 1x1.

Yes, it was a video mix of minor issues I’ve discovered so far. I forgot to show the weird sound after the Switch wakes up from standby, but I opened a specific issue about it.
Also, I’d like to add that the stash money bug (that prevents from moving gold to inventory) sometimes pops up (on Switch, minus and plus buttons open the game menu, but minus alone opens the money split window, so it’s easy to misclick it with the stash open) in a way it blocks the game entirely: the system keyboard isn’t invoked so there’s no way to close it (game and music still run, but you can’t do anything). Since it has the focus, the game can’t be controlled anymore, even menu. The only solution is to manually close the application.
However I still didn’t understand why sometimes the keyboard doesn’t appears and sometimes it does.
This is a different behavior than the inventory split gold window: that one doesn’t have the focus and calling up the menu, dismisses it.

> The swapped A and B is already a known issue.

Yes, I mentioned it in the video because I kept pressing the wrong buttons. :)

The last tiny tiny issues are that “exit game” doesn’t really quit the application (as DevilutionX does on Mac, for example), but it just replicates the “New game” command. And the “game saved” message is in English despite the different language selected.

Thanks. :)

**Comment by AJenbo** [comment]
I don't belive Switch has a way for programs to shutdown.

**Comment by locksoft** [comment]
Hmmm… maybe you’re right. :)

**Comment by locksoft** [comment]
Here's an update on this issue, with a new video. Sorry if I keep to link it instead of upload it but I get a "fail" error if I do:

http://gofile.me/6RvCQ/UcAbtLSz2

As you can see, while in games when you press the "done" button (or "+") the keyboard goes away and the typed value is passed to the game, here I need to press it twice, and it's not normal.
I show to ways in the video: in the first one you can see the value added to the prompt window for a fraction of a second and then everything closes; in the second the value is correctly passed to the window but then there's no way to "submit" as the game is expecting.

I think that whent the value is passed from the keyboard to the prompt video, this should immediately close (as if the player pressed the keyboard on the keyboard).
